### DISCUSSION TITLE
[5 giugno 2023] Esercizio 3

### DISCUSSION THREAD

**Discussion by FeddyLix17** [sapienzastudentsnetwork/psd2223#247] [Esami]
Analizzare la rete sequenziale mostrata in figura.

Stendere la tavola degli stati futuri e di uscita e disegnare l’automa (il diagramma di transizione degli stati).

<div align="center">

![image](https://github.com/user-attachments/assets/ae514708-1447-4e8b-b39a-ad4741eb6031)

</div>

**Comment by FeddyLix17** [comment]
In assenza di specifiche in merito 

- il reset asincrono può  essere [trascurato](https://t.me/c/1479650983/1/7659)

- seguendo la falsariga di una soluzione ufficiale a un appello [successivo](https://github.com/sapienzastudentsnetwork/progettazione-di-sistemi-digitali/discussions/260#:~:text=di%20Moore%20equivalente.-,soluzione%20A%2DL,-0), l'ordine da sinistra verso destra dei seguenti flip flop rappresenteranno rispettivamente $\Large S_{0:2}$

$$
    \Large
        \begin{equation}
            S_0' = \overline{A} \quad \quad
            S_1' = S_0A \quad \quad
            S_2' = A(S_2+S_1) \quad \quad
            Q = S_2
        \end{equation}
$$

<div align = "center">
                        <table>
                            <thead>
                                <tr>
                                    <th colspan="4"> Stati </th>
                                </tr>
                                <tr>
                                    <td></th>
                                    <td>$\Large S_2$</td>
                                    <td>$\Large S_1$</td>
                                    <td>$\Large S_0$</td>
                                </tr>
                            </thead>
                            <tbody>
                                <tr>
                                    <th>R</th>
                                    <td align="center">0</td>
                                    <td align="center">0</td>
                                    <td align="center">0</td>
                                </tr>
                                <tr>
                                    <th>S1</th>
                                    <td align="center">0</td>
                                    <td align="center">0</td>
                                    <td align="center">1</td>
                                </tr>
                                <tr>
                                    <th>S2</th>
                                    <td align="center">0</td>
                                    <td align="center">1</td>
                                    <td align="center">0</td>
                                </tr>
                                <tr>
                                    <th>S3</th>
                                    <td align="center">1</td>
                                    <td align="center">0</td>
                                    <td align="center">0</td>
                                </tr>
                            </tbody>
                        </table>

<table>
    <tr>
    <th colspan="16"> Tabella degli stati futuri e di uscita </th>
    </tr>
    <tr>
        <th>CS</th>
        <th rowspan="31"> </th>
        <th>$\Large S_2$</th>
        <th>$\Large S_1$</th>
        <th>$\Large S_0$</th>
        <th rowspan="31"> </th>
        <th>A</th>
        <th rowspan="31"> </th>
        <th>NS</th>
        <th rowspan="31"> </th>
        <th>$\Large S_2'$</th>
        <th>$\Large S_1'$</th>
        <th>$\Large S_0'$</th>
        <th rowspan="31"> </th>
        <th>Q</th>
    </tr>
    <tr>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">S1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">R</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">S1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">1</td>
        <td align="center">S2</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S2</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">S1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S2</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">S3</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S3</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">S1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">1</td>
    </tr>
    <tr>
        <td colspan="15"> </td>
    </tr>
    <tr>
        <td align="center">S3</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
        <td align="center">S3</td>
        <td align="center">1</td>
        <td align="center">0</td>
        <td align="center">0</td>
        <td align="center">1</td>
    </tr>
</table>
</div>

> [!NOTE]
> Sono stati omessi gli stati che non vengono mai raggiunti dalla seguente FSM, tuttavia in un compito ufficiale è bene includerli ugualmente nella tabella della verità

<div align="center"

![image](https://github.com/user-attachments/assets/345a4b86-1d85-4160-96e4-c8a82ca948c0)

</div>

**Comment by Nw4pS** [comment]
@FeddyLix17 la codifica in base a cosa è stata fatta?

**Reply by FeddyLix17** [reply]
S3 è codificato come 100 invece che 011 in quanto quest'ultima sequenza (cosi come le sequenze che vanno da 101 a 111) non vengono mai raggiunte da questa FSM.

**Reply by Nw4pS** [reply]
quindi andrebbe fatta la tabella della verità per sapere quali sono gli stati che S_0, S_1 S_2 possono assumere e poi fare la codifica
